adapters/httpapi_winhttp.c

```diff
@@ -740,6 +818,52 @@ HTTPAPI_RESULT HTTPAPI_SetOption(HTTP_HANDLE handle, const char* optionName, con
             /*winhttp accepts all certificates, because it actually relies on the system ones*/
             result = HTTPAPI_OK;
         }
+        else if (strcmp(OPTION_HTTP_PROXY, optionName) == 0)
+        {
+            char proxy[MAX_HOSTNAME_LEN];
+            HTTP_PROXY_OPTIONS* proxy_data = (HTTP_PROXY_OPTIONS*)value;
+            if (sprintf_s(proxy, MAX_HOSTNAME_LEN, "%s:%d", proxy_data->host_address, proxy_data->port) <= 0)
+            {
+                LogError("failure constructing proxy address");
+                result = HTTPAPI_ERROR;
+            }
+            else
+            {
+                httpHandleData->proxy_host = (char*)malloc(strlen(proxy)+1);
+                if(httpHandleData->proxy_host == NULL)
+                {
+                    LogError("failure allocate proxy host");
+                    result = HTTPAPI_ERROR;
+                }
+                else
+                {
+                    (void)strcpy(httpHandleData->proxy_host, proxy);
+                    if (proxy_data->username != NULL && proxy_data->password != NULL)
+                    {
+                        httpHandleData->proxy_username = (char*)malloc(strlen(proxy_data->username)+1);
+                        httpHandleData->proxy_password = (char*)malloc(strlen(proxy_data->password)+1);
+                        if (httpHandleData->proxy_username == NULL || httpHandleData->proxy_password == NULL)
+                        {
+                            LogError("failure allocate proxy username or password");
```

Comment:
I won't leave more comments on this block, waiting for a revised code using mallocAndStrcpy_s